### DISCUSSION TITLE
Is it essential to set filetype separately for every neogit buffer?

### DISCUSSION THREAD

**Discussion by ofseed** [NeogitOrg/neogit#774] [Q&A] [answered]
I have some difficulties when working neogit with other plugins.

For example, neogit changes the `gitcommit` filetype to `NeogitCommitMessage` when committing, this behavior causes my `dropbar.nvim` could not to find the correct parser for this filetype. It's hard to find the reason because this problem involves two plugins at the same time, and it seems that Neogit changes `filetype` after the buffer was created, so some autocmds will still be triggered event it is not for Neogit buffers. My final solution is to modify my local copy and not change `gitcommit` to `NeogitCommitMessage`, and I found it works well.

There are other problems. I'm using [marks.nvim](https://github.com/chentoast/marks.nvim), which shows marks in signcolumn. This plugin adds an extra status column in `Neogit*Popup`, even if it is useless. Due to its implementation, it's hard to make marks.nvim ignore by buffertype, so I must add a long list containing `NeogitLogPopup`, `NeogitStashPopup` etc. to ignore all these filetypes.

So it seems to me that there is no benefit to setting filetype separately for every buffer, for popup windows like `NeogitLogPopup`, `NeogitStashPopup`, using a unified `NeogitPopup` may be enough.

**Comment by CKolkey** [comment, marked as answer]
I've gone back and forth on this, actually. With regard to the commit message, or git rebase, buffers, for some time I had actually just changed them to be the proper "built in" filetype. This worked fine, but after searching through github found many people who had customized the `Neogit*` filetypes and... I didn't want to break that. 

It also means we can do extra stuff to the buffers without worrying about messing with another namespace. As you pointed out, this is a trade off. 

I'd recommend opening a PR on the plugins you found conflict with and add support for the neogit filetypes. That said... I don't actually see a reason that each popup be it's own filetype. I think they could totally just be `NeogitPopup` without causing any issue. I'll look into that and see if it can be done without incurring any side effects.
